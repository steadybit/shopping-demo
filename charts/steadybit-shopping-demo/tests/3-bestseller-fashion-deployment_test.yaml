templates:
  - 3-bestseller-fashion-deployment.yml
chart:
  appVersion: v0.0.0
tests:
  - it: manifest should match snapshot
    asserts:
      - matchSnapshot: { }
  - it: manifest should match snapshot replica count
    set:
      bestsellerFashion:
        replicaCount: 2
    asserts:
      - matchSnapshot: { }