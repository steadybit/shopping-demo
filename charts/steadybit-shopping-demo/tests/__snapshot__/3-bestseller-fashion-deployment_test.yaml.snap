manifest should match snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        domain: shop-products
        run: fashion-bestseller
        service-tier: "2"
        tags.datadoghq.com/service: shop-products
        tags.datadoghq.com/version: 1.0.0
        tags.steadybit.com/service-validation: http
      name: fashion-bestseller
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          run: fashion-bestseller-exposed
      template:
        metadata:
          labels:
            domain: shop-products
            run: fashion-bestseller-exposed
            service-tier: "2"
            tags.datadoghq.com/service: shop-products
            tags.datadoghq.com/version: 1.0.0
        spec:
          containers:
            - env:
                - name: DD_SERVICE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/service']
                - name: DD_VERSION
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/version']
                - name: rest.endpoint.inventory
                  value: http://inventory:8084/inventory
              image: steadybit/bestseller-fashion:develop
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /actuator/health/liveness
                  port: 8082
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 3
              name: fashion-bestseller
              ports:
                - containerPort: 8082
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /actuator/health/readiness
                  port: 8082
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 5
              resources:
                limits:
                  cpu: 500m
                  ephemeral-storage: 500Mi
                  memory: 1024Mi
                requests:
                  cpu: 100m
                  ephemeral-storage: 100Mi
                  memory: 128Mi
          serviceAccountName: steadybit-demo
  2: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        run: fashion-bestseller
      name: fashion-bestseller
      namespace: NAMESPACE
    spec:
      ports:
        - port: 8082
          protocol: TCP
          targetPort: 8082
      selector:
        run: fashion-bestseller-exposed
      type: ClusterIP
  3: |
    apiVersion: policy/v1
    kind: PodDisruptionBudget
    metadata:
      name: fashion-bestseller-pdb
      namespace: NAMESPACE
    spec:
      minAvailable: 1
      selector:
        matchLabels:
          run: fashion-bestseller-exposed
manifest should match snapshot replica count:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        domain: shop-products
        run: fashion-bestseller
        service-tier: "2"
        tags.datadoghq.com/service: shop-products
        tags.datadoghq.com/version: 1.0.0
        tags.steadybit.com/service-validation: http
      name: fashion-bestseller
      namespace: NAMESPACE
    spec:
      replicas: 2
      selector:
        matchLabels:
          run: fashion-bestseller-exposed
      template:
        metadata:
          labels:
            domain: shop-products
            run: fashion-bestseller-exposed
            service-tier: "2"
            tags.datadoghq.com/service: shop-products
            tags.datadoghq.com/version: 1.0.0
        spec:
          containers:
            - env:
                - name: DD_SERVICE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/service']
                - name: DD_VERSION
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/version']
                - name: rest.endpoint.inventory
                  value: http://inventory:8084/inventory
              image: steadybit/bestseller-fashion:develop
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /actuator/health/liveness
                  port: 8082
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 3
              name: fashion-bestseller
              ports:
                - containerPort: 8082
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /actuator/health/readiness
                  port: 8082
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 5
              resources:
                limits:
                  cpu: 500m
                  ephemeral-storage: 500Mi
                  memory: 1024Mi
                requests:
                  cpu: 100m
                  ephemeral-storage: 100Mi
                  memory: 128Mi
          serviceAccountName: steadybit-demo
  2: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        run: fashion-bestseller
      name: fashion-bestseller
      namespace: NAMESPACE
    spec:
      ports:
        - port: 8082
          protocol: TCP
          targetPort: 8082
      selector:
        run: fashion-bestseller-exposed
      type: ClusterIP
  3: |
    apiVersion: policy/v1
    kind: PodDisruptionBudget
    metadata:
      name: fashion-bestseller-pdb
      namespace: NAMESPACE
    spec:
      minAvailable: 1
      selector:
        matchLabels:
          run: fashion-bestseller-exposed
